include/pybind11/embed.h

```diff
@@ -97,6 +97,12 @@ struct wide_char_arg_deleter {
     }
 };
 
+// warning C4996: 'mbstowcs': This function or variable may be unsafe.
+#if defined(_MSC_VER)
```

Comment:
This is also only needed for Py2, correct?

Reply:
This disable warning is needed whenever mbstowcs is called. Which, according to the code, is when PY_VERSION_HEX < 0x030500f0 and HAVE_BROKEN_MBSTOWCS is not defined.

Reply:
Great, then could that be added here? Though I'd just put `PY_VERSION_HEX < 0x030500f0`. That way we will remember to remove it at the end of the year when we drop Python 2 & 3.5 support.

Reply:
Nice solution to this one. :)

Reply:
From what I see it's not gated on `PY_VERSION_HEX` or the value of `_MSC_VER`. I missed why not and why Henry wrote "Nice solution". A terse comment to explain would be helpful.

Reply:
No additional gates are needed here. Whenever we call mbstowcs function on MSVC stdlib, we need to (temporarily) disable "warning C4996: 'mbstowcs': This function or variable may be unsafe".